### PR TITLE
Work in progress on templates

### PR DESCRIPTION
**Pull request by aespasa**
The container is now sent through the Abstract Controller, in this way we can retrieve a specific instance from it where we need it (example: Templating and PathExtension)

Resolve few phpstan errors

### CODE REVIEW

File: src/Templating/Templating.php [hunk 1]
```diff
@@ -43,6 +51,7 @@ public function __construct(
         $this->twig->addExtension($this->getWebpackEncoreExtension());
         $this->twig->addExtension(new DebugExtension());
         $this->twig->addExtension(new CsrfTokenExtension());
+        $this->twig->addExtension(new PathExtension($this->container->get(Router::class)));
```
Comment (TBoileau) [inline src/Templating/Templating.php:54]: Tester : `$this->twig->addExtension($this->container->get(PathExtension::class));`

File: webpack.config.js [hunk 2]
```diff
@@ -23,6 +23,7 @@ Encore
     .addEntry('app', './assets/app.js')
 
     .addEntry('home', './assets/pages/home.js')
+    .addEntry('single', './assets/pages/single.js')
```
Comment (acecconato) [inline webpack.config.js:26]: Codacy has a fix for the issue: [Replace `····.addEntry('single',·'./assets/pages/single.js'` with `··.addEntry("single",·"./assets/pages/single.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
  .addEntry("single", "./assets/pages/single.js")
```

Comment (acecconato) [inline webpack.config.js:26]: Codacy has a fix for the issue: [Replace `····.addEntry('single',·'./assets/pages/single.js'` with `··.addEntry("single",·"./assets/pages/single.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
  .addEntry("single", "./assets/pages/single.js")
```

File: assets/app.js [hunk 3]
```diff
@@ -1,2 +1,4 @@
 // All things loaded by this entrypoint is available everywhere
+import './js/vendor/theme.js';
```
Comment (acecconato) [inline assets/app.js:2]: Codacy has a fix for the issue: [Replace `'./js/vendor/theme.js'` with `"./js/vendor/theme.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
import "./js/vendor/theme.js";
```

File: webpack.config.js (hunk 2 above)
Comment (acecconato) [inline webpack.config.js:26]: Codacy has a fix for the issue: [Replace `····.addEntry('single',·'./assets/pages/single.js'` with `··.addEntry("single",·"./assets/pages/single.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
  .addEntry("single", "./assets/pages/single.js")
```

Comment (acecconato) [inline webpack.config.js:26]: Codacy has a fix for the issue: [Replace `····.addEntry('single',·'./assets/pages/single.js'` with `··.addEntry("single",·"./assets/pages/single.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
  .addEntry("single", "./assets/pages/single.js")
```

File: assets/app.js (hunk 3 above)
Comment (acecconato) [inline assets/app.js:2]: Codacy has a fix for the issue: [Replace `'./js/vendor/theme.js'` with `"./js/vendor/theme.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
import "./js/vendor/theme.js";
```

Comment (acecconato) [inline assets/app.js:2]: Codacy has a fix for the issue: [Replace `'./js/vendor/theme.js'` with `"./js/vendor/theme.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
import "./js/vendor/theme.js";
```

File: webpack.config.js (hunk 2 above)
Comment (acecconato) [inline webpack.config.js:26]: Codacy has a fix for the issue: [Replace `····.addEntry('single',·'./assets/pages/single.js'` with `··.addEntry("single",·"./assets/pages/single.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
  .addEntry("single", "./assets/pages/single.js")
```

File: assets/app.js [hunk 4]
```diff
@@ -1,2 +1,6 @@
 // All things loaded by this entrypoint is available everywhere
-import './scss/main.scss';
\ No newline at end of file
+import './js/vendor/theme.js';
```
Comment (acecconato) [inline assets/app.js:2]: Codacy has a fix for the issue: [Replace `'./js/vendor/theme.js'` with `"./js/vendor/theme.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
import "./js/vendor/theme.js";
```

File: webpack.config.js (hunk 2 above)
Comment (acecconato) [inline webpack.config.js:26]: Codacy has a fix for the issue: [Replace `····.addEntry('single',·'./assets/pages/single.js'` with `··.addEntry("single",·"./assets/pages/single.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
  .addEntry("single", "./assets/pages/single.js")
```

Comment (aespasa) [inline webpack.config.js]: Codacy has a fix for the issue: [Replace `····.addEntry('single',·'./assets/pages/single.js'` with `··.addEntry("single",·"./assets/pages/single.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
  .addEntry("single", "./assets/pages/single.js")
```

File: assets/app.js (hunk 4 above)
Comment (aespasa) [inline assets/app.js]: Codacy has a fix for the issue: [Replace `'./js/vendor/theme.js'` with `"./js/vendor/theme.js"`](https://app.codacy.com/gh/acecconato/AnthonyCecconato_5_18102021/pullRequest?prid=9466078)
```suggestion
import "./js/vendor/theme.js";
```

**Pull request merged**